templates/cluster-template-clusterclass.yaml

```diff
@@ -698,10 +722,30 @@ spec:
               tls-cipher-suites: ${TLS_CIPHER_SUITES=TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256}
         postKubeadmCommands:
         - echo export KUBECONFIG=/etc/kubernetes/admin.conf >> /root/.bashrc
+        - |
+          KUBERNETES_VERSION_NO_V=${KUBERNETES_VERSION#v}
+          VERSION_TO_COMPARE=1.29.0
+          if [ "$(printf '%s\n' "$KUBERNETES_VERSION_NO_V" "$VERSION_TO_COMPARE" | sort -V | head -n1)" != "$KUBERNETES_VERSION_NO_V" ]; then
+            if [ -f /run/kubeadm/kubeadm.yaml ]; then
+              sed -i 's#path: /etc/kubernetes/super-admin.conf#path: /etc/kubernetes/admin.conf#' /etc/kubernetes/manifests/kube-vip.yaml;
+            fi
+          fi
         - echo "after kubeadm call" > /var/log/postkubeadm.log
         preKubeadmCommands:
         - echo "before kubeadm call" > /var/log/prekubeadm.log
         - hostnamectl set-hostname "{{ ds.meta_data.hostname }}"
+        - echo "::1         ipv6-localhost ipv6-loopback" >/etc/hosts
+        - echo "127.0.0.1   localhost" >>/etc/hosts
+        - echo "127.0.0.1   kubernetes" >>/etc/hosts
+        - echo "127.0.0.1   {{ ds.meta_data.hostname }}" >> /etc/hosts
+        - |
+          KUBERNETES_VERSION_NO_V=${KUBERNETES_VERSION#v}
+          VERSION_TO_COMPARE=1.29.0
+          if [ "$(printf '%s\n' "$KUBERNETES_VERSION_NO_V" "$VERSION_TO_COMPARE" | sort -V | head -n1)" != "$KUBERNETES_VERSION_NO_V" ]; then
+            if [ -f /run/kubeadm/kubeadm.yaml ]; then
+              sed -i 's#path: /etc/kubernetes/admin.conf#path: /etc/kubernetes/super-admin.conf#' /etc/kubernetes/manifests/kube-vip.yaml;
+            fi
+          fi
```

Comment:
Why do we need to do this in both pre and post kubeadm?

Reply:
in pre we are changing it to super-admin.conf and in post we are changing it back to admin.conf since super-admin.conf only exists on the first control plane node.